src/shared/api/LazyCache.ts

```diff
@@ -1,27 +1,34 @@
-import {observable, computed} from '../../../node_modules/mobx/lib/mobx';
+import {observable, action} from '../../../node_modules/mobx/lib/mobx';
 
 export default class LazyCache {
-    @observable.ref private _cache:any;
+    @observable.ref protected _cache:any;
+    protected dependencies:any[];
 
-    constructor(initialCache:any,
-                private dependencies:any,
-                private _populateCache:(cache:any, args:any, dependencies:any)=>Promise<boolean>) {
-        this._cache = initialCache;
+    constructor(..._dependencies:any[]) {
+        this._cache = {};
+        this.dependencies = _dependencies;
     }
 
     public get cache() {
         return this._cache;
     }
 
     public populate(args:any) {
-        this._populateCache(this._cache, args, this.dependencies).then((didChange:boolean) => {
+        this.populateCache(args, ...this.dependencies).then((didChange:boolean)=>{
```

Comment:
lets think about whether it would be better to just use immutable to handle this.  the data arguably should be immutable for other reasons, so why not just use it to handle this business.